### ISSUE TITLE
Problem with type reassignment

### ISSUE THREAD

**Issue by jolaf** [python/mypy#13052]
The following code:
```python3
_type = type

def f(a: type[str]) -> None:
    pass

def g(a: _type[str]) -> None:
    pass
```
produces the following output:
```shell
$ mypy test.py
test.py: note: In function "g":
test.py:6:10: error: "type" expects no type arguments, but 1 given  [type-arg]
    def g(a: _type[str]) -> None:
             ^
Found 1 error in 1 file (checked 1 source file)

```
This looks like a false positive to me.

```shell
$ mypy --version
mypy 0.961 (compiled: yes)

$ python3 --version
Python 3.10.4

$ lsb_release -d
Description:	Ubuntu 22.04 LTS
```

**Comment by erictraut** [comment]
You're defining a generic type alias called `_type` here and assigning it the type `type`, which takes one type argument. Since you haven't assigned a type argument, it defaults to `Any`, so `_type` is the equivalent of `type[Any]`, and you cannot specialize it again.

If your intent was for `_type` to remain generic, you can define the type alias as follows.

```python
T = TypeVar("T")
_type = type[T]
```

**Comment by jolaf** [comment]
[erictraut](https://github.com/erictraut) Oh, thanks, that really helped me with my particular problem!

**Comment by AlexWaygood** [comment]
Duplicate of #10068
